irony-cdb-libclang.el

```diff
@@ -0,0 +1,89 @@
+;;; irony-cdb-libclang.el --- Compilation Database for irony using libclang
+
+;; Copyright (C) 2014  Guillaume Papin
+
+;; Author: Guillaume Papin <guillaume.papin@epitech.eu>
+;; Keywords: c, convenience, tools
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;;
+;; Compilation Database support for Irony using libclangs CXCompilationDatabase.
+;; Currently libclang only supports json compilation databases, but reading
```

Comment:
I don't know about this comment, since it is not entirely true maybe we will be better of without it.

The slowness comes from:

* json.el
* the fact that the Emacs built-in functions for splitting a command line are buggy
* that I had to roll my own parsing but it's badly implemented, very slow
* ...